### PR TITLE
[GCOS] Improve support for sections and paragraphs that do not end with a period

### PR DESCRIPTION
**Pull request by nberth**

### CODE REVIEW

File: cobc/pplex.l [hunk 1]
```diff
@@ -437,7 +437,7 @@ DEFNUM_LITERAL	[+-]?[0-9]*[\.]*[0-9]+
 	return COPY;
 }
 
-"REPLACE"		{
+<*>"REPLACE"		{
```
Comment (lefessan) [inline cobc/pplex.l:440]: is it specific to GCOS or standard ?

Comment (nberth) [reply cobc/pplex.l:440]: This is to support replace statements in control divisions, so it is specific to GCOS.
This edit allows us to remove one rule below in `pplex.l`.

Comment (lefessan) [reply cobc/pplex.l:440]: My point was more about the fact that it is modified for all dialects, not only GCOS. Isn't it a problem ?

Comment (nberth) [reply cobc/pplex.l]: Hard to assess as it depends on where the various dialects allow `REPLACE` to appear, and all may not be clear about that. Also most other start conditions relate to compiler directives where the `REPLACE` word is forbidden AFAIK.

But the case may still by open for `COPY_STATE` and `PSEUDO_STATE`. In `COPY_STATE`, a `COPY` statement is scanned, and there I think "`REPLACE`" is forbidden too. For `PSEUDO_STATE`, indeed it's not much clearer…
So while in doubt I've pushed a change to explicitly restrict to previous behaviors for the other dialects.

**Pull request merged**